be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/handlers/SocialMediaHandler.scala

```diff
@@ -70,20 +70,17 @@ class SocialMediaHandler(dbRef: => AskableActorRef) extends MessageHandler {
     writeAndPropagate(rpcMessage) match {
       case Left(_) =>
         val channelChirp: Channel = rpcMessage.getParamsChannel
-        channelChirp.decodeChannelLaoId match {
-          case Some(lao_id) =>
-            val broadcastChannel: Channel = generateSocialChannel(lao_id)
-            rpcMessage.getParamsMessage match {
-              case Some(params) =>
-                // we can't get the message_id as a Base64Data, it is a Hash
-                val chirp_id: Hash = params.message_id
-                val timestamp: Timestamp = params.decodedData.get.asInstanceOf[AddChirp].timestamp
-                val notifyAddChirp: NotifyAddChirp = NotifyAddChirp(chirp_id, channelChirp, timestamp)
-                val broadcastData: Base64Data = Base64Data.encode(notifyAddChirp.toJson.toString)
-                broadcastHelper(rpcMessage, broadcastData, broadcastChannel)
-              case None => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, "Server failed to extract chirp id for the broadcast", rpcMessage.id))
-            }
-          case None => Right(PipelineError(ErrorCodes.SERVER_ERROR.id, "Server failed to extract LAO id for the broadcast", rpcMessage.id))
+        val lao_id: Hash = channelChirp.decodeChannelLaoId
```

Comment:
How do we handle the case in which the decodeChannelLaoId might fail? (Not saying it's the case here, since we are in the handler, but if it may fail elsewhere, we could need the Option return value, correct me if I'm wrong)